mitodl/mitxpro
# Add pages for bulk enrollment code purchase and a receipt page to download codes

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #858 

#### What's this PR do?
Implements the site where b2b customers can purchase enrollment codes which would be used by end users to enroll in the course run or program bundle of course runs.

#### How should this be manually tested?
First note that the functionality should work exactly the same with or without login.

For the purchase page, select a product and enter the number of seats and your email address. Click 'Place Order' and you will be directed to Cybersource. Fill in the fake credit card information and you should be redirected to the receipt page. You should see a simple "Loading..." message. If you look at the network tab you should see a request made every three seconds to the new order status API.

To test the receipt page we need to fulfill the order in the backend. You could try to fulfill the order via the order fulfillment REST API but this is a lot of work so we may want to skip this and test that on RC instead. At this point a `B2BOrder` with `status = created` should be created in the database. It's probably easiest to just set `status = fulfilled` manually, via the Django admin or shell. Once the status is updated the receipt page should automatically show the right data along with a link to download coupon codes.

Click the link to download the enrollment codes and verify that the browser sees that it's a CSV file. The CSV file should be very simple, just a list of codes.

#### Not included in this PR but will be worked on for a follow-up PR
 - email to purchaser
 - modal select of courses and programs

#### Screenshots
Purchase page
Desktop:
![purchase](https://user-images.githubusercontent.com/863262/62960795-6dd0fe00-bdc9-11e9-9e57-77902fd932d7.png)

Mobile:
![mobile_purchase](https://user-images.githubusercontent.com/863262/62960783-6a3d7700-bdc9-11e9-9cfe-5d836495d2ed.png)

Receipt page
Desktop:
![receipt](https://user-images.githubusercontent.com/863262/62960774-63166900-bdc9-11e9-9bab-064535c6e063.png)

Mobile:
![mobile_receipt](https://user-images.githubusercontent.com/863262/62960765-5d208800-bdc9-11e9-8a81-6f1765a2bb1b.png)

Error message:
![error](https://user-images.githubusercontent.com/863262/62960759-58f46a80-bdc9-11e9-931c-0378a8043b58.png)
